File: test_dir/python/src/loops/while_.py
def while_loop_fn():
    # typical while loop
    a: list[int] = []
    i: int = 0
    while i < 3:
        a.append(i)
        i += 1
    print(a)
    # break statement
    while True:
        a.append(i)
        if i > 3:
            break
        i += 1
    print(a)
    # continue statement
    while i < 7:
        if i == 5:
            i += 1
            continue
        a.append(i)
        i += 1
    print(a)

File: test_dir/cpp/src/loops/while_.cpp
#include "loops\while_.h"
#include "py_list.h"
#include "py_str.h"
#include "pypp_util/print.h"

void while_loop_fn() {
    print(PyStr("WHILE LOOP RESULTS:"));
    PyList<int> a = PyList<int>({});
    int i = 0;
    while (i < 3) {
        a.append(i);
        i += 1;
    }
    print(a);
    while (true) {
        a.append(i);
        if (i > 3) {
            break;
        }
        i += 1;
    }
    print(a);
    while (i < 7) {
        if (i == 5) {
            i += 1;
            continue;
        }
        a.append(i);
        i += 1;
    }
    print(a);
}